superset-frontend/src/components/Chart/Chart.jsx

```diff
@@ -216,7 +219,8 @@ class Chart extends React.PureComponent {
     if (
       chartAlert !== undefined &&
       chartAlert !== NONEXISTENT_DATASET &&
-      datasource === PLACEHOLDER_DATASOURCE
+      datasource === PLACEHOLDER_DATASOURCE &&
+      datasetsStatus !== ResourceStatus.ERROR
```

Comment:
I haven't worked on this component for a while, but I would have expected it to check the other way around, i.e. are the conditions met for displaying the error, and if not, only *then* display the spinner. But I don't propose changing this now as it would probably introduce unnecessary regression risk.